viam-cartographer/src/carto_facade/carto_facade.cc

```diff
@@ -627,8 +627,46 @@ int CartoFacade::Stop() {
     return VIAM_CARTO_SUCCESS;
 };
 
-int CartoFacade::AddSensorReading(viam_carto_sensor_reading *sr) {
-    return VIAM_CARTO_SUCCESS;
+void CartoFacade::AddSensorReading(const viam_carto_sensor_reading *sr) {
+    std::string sensor = to_std_string(sr->sensor);
+    if (config.sensors[0] != sensor) {
+        VLOG(1) << "expected sensor: " << sensor << " to be "
+                << config.sensors[0];
+        throw VIAM_CARTO_SENSOR_NOT_IN_SENSOR_LIST;
```

Comment:
are we expecting only one sensor? also is it possible for this to get called before config validation checks that the length of the sensor list != 0?

Reply:
1. For now yes. In the slam_service only the first element of the sensors list is provided by the viam-server json.config will ever be used. This attempts to keep the same behavior while leaving an obvious path to accommodate other sensors (such as IMU) in the future.
2. I don't think so given that the sensor list is added as part of the constructor, which will throw an exception if the sensors list is empty.https://github.com/viamrobotics/viam-cartographer/pull/156/files/4a371f514999eeb22509c413b3c3d975dd727a7a#diff-a0af44f9985e7739a638c01fc02f2c543e0f957e732561ea6bcea0dc6891a97fR158